app/controllers/application_controller.rb

```diff
@@ -5,10 +5,17 @@ class ApplicationController < ActionController::Base
   before_action :reload_extensions, unless: -> { Rails.env.production? }
   before_action -> { cookies.encrypted[:example_id] ||= SecureRandom.uuid }
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :masquerade_user!
   before_action :set_paper_trail_whodunnit
 
   protected
 
+  def authenticate_administrator!
+    unless AuthorizedUser.new(current_user).can_admin_system?
+      render status: :forbidden, text: "You do not have access to this page."
+    end
+  end
+
```

Comment:
Note that we have an "official" forbidden page. It's currently styled like other Rails error pages, but I'm expecting that we'll customize such pages to match the site. 

```ruby
return render_forbidden
```

Should we also verify that the user isn't masquerading in our admin auth?

```ruby
return render_forbidden if user_masquerade?
return render_forbidden unless AuthorizedUser.new(current_user).can_admin_system?
```

Reply:
Are you aware of the __pretender__ gem? https://github.com/ankane/pretender

I'm not sure how it stacks up against __devise_authorizer__ but it may be worth considering also? More stars, the docs are better, and they've addressed concerns like ActionCable.